web/src/navbar_menus.js

```diff
@@ -12,46 +19,85 @@ export function is_navbar_menus_displayed() {
+        i = i + 1;
     }
 
-    if (
-        popover_menus.is_help_menu_popover_displayed() &&
-        (event_name === "right_arrow" || event_name === "gear_menu")
-    ) {
-        // Open gear menu popover on right arrow.
-        navbar_help_menu.toggle();
-        gear_menu.toggle();
-        return true;
+    return false;
+}
+
+// Check if a specific menu is displayed
+function is_menu_displayed(selected_popover_menu) {
+    switch (selected_popover_menu) {
+        case "signup_button":
+            return document.activeElement.className === "signup_button";
+        case "login_button":
+            return document.activeElement.className === "login_button";
+        case "navbar_help_menu":
+            return popover_menus.is_help_menu_popover_displayed();
+        case "gear_menu":
+            return popover_menus.is_gear_menu_popover_displayed();
+        case "personal_menu":
+            return popover_menus.is_personal_menu_popover_displayed();
     }
+    return false;
+}
 
-    if (popover_menus.is_gear_menu_popover_displayed()) {
-        if (event_name === "gear_menu") {
-            gear_menu.toggle();
+// Handle navigation events for specific menus
+function handle_menu_event(selected_popover_menu, event_name, index) {
+    switch (event_name) {
+        case "gear_menu":
+            toggle_menu(selected_popover_menu);
             return true;
-        } else if (event_name === "right_arrow" && !page_params.is_spectator) {
-            // Open personal menu popover on g + right arrow.
+        case "left_arrow":
+            if (index > 0) {
+                toggle_menu(selected_popover_menu);
+                toggle_menu(popover_menus_options[index - 1]);
+                return true;
+            }
+            break;
+        case "right_arrow":
+            if (index < popover_menus_options.length - 1) {
+                toggle_menu(selected_popover_menu);
+                toggle_menu(popover_menus_options[index + 1]);
+                return true;
+            }
+            break;
+    }
+    return false;
+}
+
+// Toggle the display state of a specific menu
+function toggle_menu(selected_popover_menu) {
+    switch (selected_popover_menu) {
+        case "signup_button":
+        case "login_button":
+            toggle_button_focus(selected_popover_menu);
+            break;
+        case "navbar_help_menu":
+            navbar_help_menu.toggle();
+            break;
+        case "gear_menu":
             gear_menu.toggle();
+            break;
+        case "personal_menu":
             personal_menu_popover.toggle();
-            return true;
-        } else if (event_name === "left_arrow") {
-            // Open help menu popover on g + left arrow.
-            gear_menu.toggle();
-            navbar_help_menu.toggle();
-            return true;
-        }
+            break;
     }
+}
 
-    return false;
+// Toggle the focus state of a button
+function toggle_button_focus(selected_popover_menu) {
```

Comment:
This function now also be refactored to use document.activeElement.

implementation
``` javascript
function toggle_button_focus(selected_popover_menu) {
    const selected_element = document.querySelector("." + selected_popover_menu);
    if (document.activeElement.classList.contains(selected_popover_menu)) {
        selected_element.blur();
    } else {
        selected_element.focus();
    }
}
```